### PR TITLE
refactor(radio): refactor the Radio, RadioGroup and RadioGroupField incl. new SVGs

### PR DESCRIPTION
**Pull request by varl**
Relates to #457.

This refactor will include the `Radio` (atom) component and some form of `*-Field` (molecule) component that can be used to construct radio groups.

ToDos:
- [x] Implement `Radio`
    - [x] Allow prop-type `node` in `label`
    - [x] Add `dense` prop and style
- [x] Implement `RadioGroup`
- [x] Implement `RadioGroupField`
- [x] Tweak styling of components within `RadioGroupField`
- [x] Rework `Checkbox` and `Radio` SVG (see https://github.com/dhis2/ui-core/issues/515)

### CODE REVIEW

**Comment by HendrikThePendric** [comment]
When I implemented the `RadioGroupField` (see https://github.com/dhis2/ui-core/pull/458/commits/1ce48e54d7c50393cf31570c7679b09ae37c0343) I noticed some imperfections reg. the styles. @cooper-joe could you review the following points:
- Spacing with label: The distance between the groups seems to be equal to the spacing between the legend/label and the checkboxes. I don't think this looks right.
- Dense mode: The radios themselves support dense mode. However, the legend/label and the help/error texts don't. Is that OK? Or should we at least add a dense mode to the Legend?
- Disabled mode: The Help texts change color for error/warning/valid state, but not for disabled. Is that OK?

The deploy preview link is up-to-date so you can use that to review the points above.

**Comment by HendrikThePendric** [comment]
Something else I noticed just now, but haven't been able to find an elegant solution for yet:
- The width of the `label` that wraps the radio-input is 100%. You could argue this is a feature, but quite often this is confusing.
- The most logical way to control this, in theory, would be to apply the `display: flex;` property to the `FieldSet` component....
- But as @Mohammer5 pointed out at some point, `fieldset`s can't be flex containers. So we'd have to add a div inside or around the fieldset to get this behavior to work...
- Adding `display: flex;` to the `Field` component fixes things too, but that doesn't make as much sense conceptually, and also breaks some other Fields. I suppose we could add it as a prop  to the `Field` component?

Any other suggestions on how to fix this? And/or opinions on whether this is a bug or a feature?
@ismay @cooper-joe @Mohammer5 @varl

**Comment by HendrikThePendric** [comment]
**UPDATE:** probably just applying `display: inline-block` to the `FieldSet` as I did in https://github.com/dhis2/ui-core/pull/458/commits/2e815b572d61153a94ab6afea0ff54f3ebd9b2bc is the most appropriate solution.

**Comment by cooper-joe** [comment]
> When I implemented the `RadioGroupField` (see [1ce48e5](https://github.com/dhis2/ui-core/commit/1ce48e54d7c50393cf31570c7679b09ae37c0343)) I noticed some imperfections reg. the styles. @cooper-joe could you review the following points:
> 
>     * Spacing with label: The distance between the groups seems to be equal to the spacing between the legend/label and the checkboxes. I don't think this looks right.

Spacing for groups added to [updated spec](https://sketch.cloud/s/DwkDk/a/Ak9J0M)

>     * Dense mode: The radios themselves support dense mode. However, the legend/label and the help/error texts don't. Is that OK? Or should we at least add a dense mode to the Legend?

Dense mode labels/help text are the same as regular mode. Dense mode controls are actually intended more for usage in UI components where help/label/states won't be used, so this isn't a issue (right now).

>     * Disabled mode: The Help texts change color for error/warning/valid state, but not for disabled. Is that OK?

Yes, that's fine.

**Comment by Mohammer5** [comment]
> The width of the label that wraps the radio-input is 100%. You could argue this is a feature, but quite often this is confusing.

I think this is fine, they're behaving like block elements so they're not inline.
The fieldset with it's initial display value (which is `block` if I'm not mistaken) adjusts to it's parent's `width`, which is fine as well. If the UI requires different behavior, it's up to the app.
So I think there's no need for this:

> UPDATE: probably just applying display: inline-block to the FieldSet as I did in 2e815b5 is the most appropriate solution.

File: src/FieldSet.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ const FieldSet = ({ className, children }) => (
                 border: none;
                 margin: 0;
                 padding: 0;
+                display: inline-block;
```
Comment (varl) [inline src/FieldSet.js]: A fieldset should be a block element.

File: src/RadioGroupField.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+            {helpText && <Help>{helpText}</Help>}
+
+            {validationText && (
+                <Help error={error} warning={warning} valid={valid}>
+                    {validationText}
+                </Help>
+            )}
+        </FieldSet>
+    </Field>
```
Comment (varl) [inline src/RadioGroupField.js]: Nice!

File: src/FieldSet.js (hunk 1 above)
Comment (varl) [reply src/FieldSet.js]: c7e7afa

**Comment by varl** [comment]
@HendrikThePendric I don't see this point addressed in this PR:

>I don't think we need to merge branch #514 that @ismay worked on. Instead I'll just cherry-pick e98f744

**Comment by ismay** [comment]
@varl I assume that that's something that will be addressed in #516. Or is that not what you meant?

**Comment by varl** [comment]
> 
> 
> @varl I assume that that's something that will be addressed in #516. Or is that not what you meant?

Yeah, I mixed them up.

**Comment by varl** [comment]
This looks good!

Please **squash locally**[1] and add a conventional commit message before merging.

Remember that **if this is a breaking change**[2] you need to add a footer to the commit message with the words `BREAKING CHANGE:` and a list of the changes that are breaking.

[1] There are many articles on `git rebase -i`, here's one: https://stackabuse.com/git-squash-multiple-commits-in-to-one-commit/
[2] https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

**Comment by HendrikThePendric** [comment]
@varl I've tried following your instruction reg. squashing commits etc. With @ismay's help I think I've achieved the correct result. I believe it was mentioned somewhere that it would be good to keep the original commit messages in the squashed commit message, so I've done that too.

Could you let me know if this is OK, and if so, approve this PR?

**Pull request merged**